[CI] RoundToTests testCrankyEvaluateBlockWithoutNulls {TestCase=<int, 20 fixed-interval ints>} failing

**Build Scans:**
- [elasticsearch-intake #36685 / part3](https://gradle-enterprise.elastic.co/s/icsizcb6bzva6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.scalar.math.RoundToTests.testCrankyEvaluateBlockWithoutNulls {TestCase=<int, 20 fixed-interval ints>}" -Dtests.seed=F4FA0DA522A19FAA -Dtests.locale=kxv-Orya -Dtests.timezone=Europe/Gibraltar -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.expression.function.scalar.math.RoundToTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCrankyEvaluateBlockWithoutNulls%20%7BTestCase%3D%3Cint%2C%2020%20fixed-interval%20ints%3E%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <800>
     but: was <900>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testCrankyEvaluateBlockWithoutNulls {TestCase=<int, 20 fixed-interval ints>}
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -369,6 +369,30 @@ tests:
 - class: org.elasticsearch.xpack.sql.qa.jdbc.single_node.SingleNodeJdbcResultSetIT
   method: testGettingValidShortWithoutCasting
   issue: https://github.com/elastic/elasticsearch/issues/145241
+- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
+  method: test {csv-spec:approximation.Approximated column and null}
+  issue: https://github.com/elastic/elasticsearch/issues/145248
+- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
+  method: test {csv-spec:approximation.Warn on no stats}
+  issue: https://github.com/elastic/elasticsearch/issues/145249
+- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
+  method: test {csv-spec:approximation.Exact stats by on small multi-valued data}
+  issue: https://github.com/elastic/elasticsearch/issues/145250
+- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
+  method: test {csv-spec:approximation.Exact stats on small multi-valued data}
+  issue: https://github.com/elastic/elasticsearch/issues/145251
+- class: org.elasticsearch.xpack.esql.expression.function.scalar.math.RoundToTests
+  method: testFold {TestCase=<int, 15 fixed-interval ints>}
+  issue: https://github.com/elastic/elasticsearch/issues/145254
+- class: org.elasticsearch.xpack.esql.expression.function.scalar.math.RoundToTests
+  method: testFold {TestCase=<long, 15 fixed-interval longs>}
+  issue: https://github.com/elastic/elasticsearch/issues/145255
+- class: org.elasticsearch.xpack.esql.expression.function.scalar.math.RoundToTests
+  method: testEvaluateInManyThreads {TestCase=<long, 11 fixed-interval longs>}
+  issue: https://github.com/elastic/elasticsearch/issues/145256
+- class: org.elasticsearch.xpack.restart.FullClusterRestartIT
+  method: testWatcherWithApiKey {cluster=UPGRADED}
+  issue: https://github.com/elastic/elasticsearch/issues/131964
 
 # Examples:
 #

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -369,9 +369,6 @@ tests:
 - class: org.elasticsearch.xpack.sql.qa.jdbc.single_node.SingleNodeJdbcResultSetIT
   method: testGettingValidShortWithoutCasting
   issue: https://github.com/elastic/elasticsearch/issues/145241
-- class: org.elasticsearch.xpack.esql.expression.function.scalar.math.RoundToTests
-  method: testCrankyEvaluateBlockWithoutNulls {TestCase=<int, 20 fixed-interval ints>}
-  issue: https://github.com/elastic/elasticsearch/issues/145245
 
 # Examples:
 #

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/RoundToTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/RoundToTests.java
@@ -210,7 +210,7 @@ public class RoundToTests extends AbstractScalarFunctionTestCase {
         return supplier(name, fieldType, fieldSupplier, pointsTypes, pointsSupplier, (f, p) -> {
             long max = p.stream().mapToLong(l -> l).min().getAsLong();
             for (long l : p) {
-                if (l > max && f.doubleValue() > l) {
+                if (l > max && f.doubleValue() >= l) {
                     max = l;
                 }
             }
@@ -228,7 +228,7 @@ public class RoundToTests extends AbstractScalarFunctionTestCase {
         return supplier(name, fieldType, fieldSupplier, pointsTypes, pointsSupplier, (f, p) -> {
             int max = p.stream().mapToInt(i -> i).min().getAsInt();
             for (int l : p) {
-                if (l > max && f.doubleValue() > l) {
+                if (l > max && f.doubleValue() >= l) {
                     max = l;
                 }
             }
@@ -263,7 +263,7 @@ public class RoundToTests extends AbstractScalarFunctionTestCase {
         return supplier(name, fieldType, fieldSupplier, pointsTypes, pointsSupplier, (f, p) -> {
             double max = p.stream().mapToDouble(d -> d).min().getAsDouble();
             for (double d : p) {
-                if (d > max && f.doubleValue() > d) {
+                if (d > max && f.doubleValue() >= d) {
                     max = d;
                 }
             }
@@ -281,7 +281,7 @@ public class RoundToTests extends AbstractScalarFunctionTestCase {
         return supplier(name, fieldType, fieldSupplier, pointsTypes, pointsSupplier, (f, p) -> {
             long max = p.stream().mapToLong(l -> l).min().getAsLong();
             for (long l : p) {
-                if (l > max && f.doubleValue() > l) {
+                if (l > max && f.doubleValue() >= l) {
                     max = l;
                 }
             }
@@ -299,7 +299,7 @@ public class RoundToTests extends AbstractScalarFunctionTestCase {
         return supplier(name, fieldType, fieldSupplier, pointsTypes, pointsSupplier, (f, p) -> {
             int max = p.stream().mapToInt(i -> i).min().getAsInt();
             for (int l : p) {
-                if (l > max && f.doubleValue() > l) {
+                if (l > max && f.doubleValue() >= l) {
                     max = l;
                 }
             }

